### PR TITLE
Add MyPy to CI

### PR DESCRIPTION
**Pull request by TommyDew42**
**Patch Description**
PR for [issue 14](https://github.com/facebookresearch/metaseq/issues/14).

Here we add MyPy to CI.

**Testing steps**
Run `mypy` at local laptop & run the check via an actual pull request.

### CODE REVIEW

**Comment by TommyDew42** [comment]
@stephenroller 

I run the CI in the [PR](https://github.com/TommyDew42/metaseq/pull/1) in my fork repo.
But there are a lot of `mypy` type checking [errors](https://github.com/TommyDew42/metaseq/runs/6474179539?check_suite_focus=true).

How do we do about them?

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -44,3 +44,8 @@ jobs:
       run: |
         pip install black
         black --check .
+
+    - name: Type Checking with MyPy
+      run:  |
+        pip install mypy
+        mypy
```
Comment (stephenroller) [inline .github/workflows/lint.yml:51]: Lets make this || true for now. Mypy is best as a warning not an error in its current state

File: mypy.ini [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[mypy]
+python_version = 3.7
```
Comment (suchenzang) [inline mypy.ini:2]: I believe we have python 3.8 dependency in our CI setup - could we change this to 3.8 instead?

**Review by suchenzang — APPROVED** [review]

**Comment by stephenroller** [comment]
Thanks for the contribution.

**Pull request merged**